### DISCUSSION TITLE
Lesson 9 - Setup failed: Custom Error d8a3fb52:(0x8Bbbb5d294c78bd70dd88E6D63498dECa0C7BBCe)

### DISCUSSION THREAD

**Discussion by corps21** [Cyfrin/foundry-full-course-cu#947] [Q&A] [answered]
I was running local test cases, which is resulting in custom error, the error is pointing to AddConsumer contract, Here is the error
![Screenshot 2023-10-01 123439](https://github.com/Cyfrin/foundry-full-course-f23/assets/46529877/05cd1d0b-c64d-451f-850f-443a131768ab)

here is the AddConsumer Contract
```
contract AddConsumer is Script {
    function run() external {
        address contractAddress = DevOpsTools.get_most_recent_deployment("Lottery", block.chainid);
        addConsumerUsingConfig(contractAddress);
    }

    function addConsumerUsingConfig(address _contractAddress) internal {
        HelperConfig helperConfig = new HelperConfig();
        (,, address vrfCoordinatorV2,, uint64 subId,,) = helperConfig.activeNetworkConfig();
        addConsumer(_contractAddress, vrfCoordinatorV2, subId);
    }

    function addConsumer(address _contractAddress, address vrfCoordinatorV2, uint64 subId) public {
        vm.startBroadcast();
        VRFCoordinatorV2Mock(vrfCoordinatorV2).addConsumer(subId, _contractAddress);
        vm.stopBroadcast();
    }

}
```

here is the DeployLottery Contract 

```
contract DeployLottery is Script {
    function run() external returns (Lottery, HelperConfig) {
        HelperConfig helperConfig = new HelperConfig();
        (
            uint256 lotteryFee,
            uint256 interval,
            address vrfcoordinator,
            bytes32 keyHash,
            uint64 subscriptionId,
            uint32 callbackGasLimit,
            address link
        ) = helperConfig.activeNetworkConfig();

        if (subscriptionId == 0) {
            CreateSubscription createSubscription = new CreateSubscription();
            subscriptionId = createSubscription.createSubscriptionId(vrfcoordinator);
            
            FundSubscription fundSubscriptions = new FundSubscription();
            fundSubscriptions.fundSubscription(vrfcoordinator, subscriptionId, link);
        }


        vm.startBroadcast();
        Lottery lottery = new Lottery(
                lotteryFee,
                interval,
                vrfcoordinator,
                keyHash,
                subscriptionId,
                callbackGasLimit
            );
        vm.stopBroadcast();

        AddConsumer addConsumers = new AddConsumer();
        addConsumers.addConsumer(address(lottery), vrfcoordinator, subscriptionId);

        return (lottery, helperConfig);
    }
}
```

Here is my [repo](https://github.com/corps21/Patricks-Foundry-Course)
Help will be appreciated  😊😊😊

**Comment by theirrationalone** [comment, marked as answer]
You're just omitting that `deployerKey` aka `privateKey` stuff. 

Please use `vm.envUint("privateKey")` for sepolia and make a `uint256 deployerKey = hardcode_any_anvil_privateKey` for anvil. Add them to their respective places into helperConfig.sol file and sure to also return them with returning data. 

After adding deployerKey into helperConfig.sol file you can retrieve it wherever you need it. 

Now you have to pass this deployerKey to all places where you used `vm.startBroadcast()`with the addition of deployerKey as an arg into it. For example: `vm.startBroadcast(deployerKey)`.

**Reply by corps21** [reply]
Thanks, my issue is now resolved, but how did Patrick's code pass all the without deployerKey?

**Reply by theirrationalone** [reply]
That was a limited lesson upto knowledge. You implemented so much before patrick. You can verify by comparing yours with patrick's limited scripts code implementation and tests performed against.

**Reply by DukeLurkin** [reply]
I had this exact problem.  deployerKey is there in his repo code but I'm almost positive he must've accidentally skipped over it in the video.
